dsl/src/main/scala/com.crobox.clickhouse/dsl/column/ArithmeticFunctions.scala

```diff
@@ -0,0 +1,118 @@
+  implicit object DoubleFloatBinding extends AritRetType[Double,Float,Float]
+  implicit object DoubleBigDecimalBinding extends AritRetType[Double,BigDecimal,BigDecimal]
+  implicit object DoubleBigIntBinding extends AritRetType[Double,BigInt,BigDecimal]
+
+  implicit object FloatIntBinding extends AritRetType[Float,Int,Float]
+  implicit object FloatLongBinding extends AritRetType[Float,Long,Float]
+  implicit object FloatDoubleBinding extends AritRetType[Float,Double,Float]
+  implicit object FloatFloatBinding extends AritRetType[Float,Float,Float]
+  implicit object FloatBigDecimalBinding extends AritRetType[Float,BigDecimal,BigDecimal]
+  implicit object FloatBigIntBinding extends AritRetType[Float,BigInt,BigDecimal]
+
+  implicit object BigDecimalIntBinding extends AritRetType[BigDecimal,Int,BigDecimal]
+  implicit object BigDecimalLongBinding extends AritRetType[BigDecimal,Long,BigDecimal]
+  implicit object BigDecimalDoubleBinding extends AritRetType[BigDecimal,Double,BigDecimal]
+  implicit object BigDecimalFloatBinding extends AritRetType[BigDecimal,Float,BigDecimal]
+  implicit object BigDecimalBigDecimalBinding extends AritRetType[BigDecimal,BigDecimal,BigDecimal]
+  implicit object BigDecimalBigIntBinding extends AritRetType[BigDecimal,BigInt,BigDecimal]
+
+  implicit object BigIntIntBinding extends AritRetType[BigInt,Int,BigInt]
+  implicit object BigIntLongBinding extends AritRetType[BigInt,Long,BigInt]
+  implicit object BigIntDoubleBinding extends AritRetType[BigInt,Double,BigDecimal]
+  implicit object BigIntFloatBinding extends AritRetType[BigInt,Float,BigDecimal]
+  implicit object BigIntBigDecimalBinding extends AritRetType[BigInt,BigDecimal,BigDecimal]
+  implicit object BigIntBigIntBinding extends AritRetType[BigInt,BigInt,BigInt]
+
+  implicit object LocalDateIntBinding extends AritRetType[LocalDate,Int,LocalDate]
+  implicit object LocalDateLongBinding extends AritRetType[LocalDate,Long,LocalDate]
+  implicit object LocalDateDoubleBinding extends AritRetType[LocalDate,Double,LocalDate]
+  implicit object LocalDateFloatBinding extends AritRetType[LocalDate,Float,LocalDate]
+  implicit object LocalDateBigDecimalBinding extends AritRetType[LocalDate,BigDecimal,LocalDate]
+  implicit object LocalDateBigIntBinding extends AritRetType[LocalDate,BigInt,LocalDate]
+
+  implicit object DateTimeIntBinding extends AritRetType[DateTime,Int,DateTime]
+  implicit object DateTimeLongBinding extends AritRetType[DateTime,Long,DateTime]
+  implicit object DateTimeDoubleBinding extends AritRetType[DateTime,Double,DateTime]
+  implicit object DateTimeFloatBinding extends AritRetType[DateTime,Float,DateTime]
+  implicit object DateTimeBigDecimalBinding extends AritRetType[DateTime,BigDecimal,DateTime]
+  implicit object DateTimeBigIntBinding extends AritRetType[DateTime,BigInt,DateTime]
+  
+
+  def plus[L, R, O](left: AddSubtractable[L], right: AddSubtractable[R])(implicit ev: AritRetType[L,R,O])    = Plus[O](left, right)
```

Comment:
Love the fact that we can choose between `a + b` and `plus(a,b)`

Reply:
For more functions I have introduced a postfix way of doing them, check StringOps and StringSearchOps etc. These kinds of functions could be added for almost anything in the same way as we did it there.